### PR TITLE
Better sorting algorithm while searching parent pages

### PR DESCRIPTION
**Pull request by youknowriad**
This PR improves the search algorithm for parent pages by adding some client logic to:

 - Give priority to equal strings
 - followed by pages that start with the search string

### CODE REVIEW

File: packages/editor/src/components/page-attributes/parent.js [hunk 1]
```diff
@@ -29,10 +31,23 @@ function getTitle( post ) {
...
 }
 
+export const getItemPriority = ( name, searchValue ) => {
+	const normalizedName = deburr( name ).toLowerCase();
+	const normalizedSearch = deburr( searchValue ).toLowerCase();
+	if ( normalizedName === normalizedSearch ) {
+		return 0;
+	}
+
+	if ( normalizedName.indexOf( normalizedSearch ) === 0 ) {
```
Comment (ntsekouras) [inline packages/editor/src/components/page-attributes/parent.js:41]: Nitpick - we could use `String.prototype.startsWith`.

**Review by ntsekouras — APPROVED** [review]
Tested and looks good. It feels a bit weird that `getItemPriority` returned bigger numbers for items we want lower position, but overall everything seemed fine :)

File: packages/editor/src/components/page-attributes/parent.js [hunk 2]
```diff
@@ -77,25 +92,36 @@ export function PageAttributesParent() {
...
-				value: treeNode.id,
-				label: repeat( '— ', level ) + unescapeString( treeNode.name ),
-			},
-			...getOptionsFromTree( treeNode.children || [], level + 1 ),
-		] );
-	};
 
 	const parentOptions = useMemo( () => {
+		const getOptionsFromTree = ( tree, level = 0 ) => {
+			const mappedNodes = map( tree, ( treeNode ) => [
```
Comment (ZebulanStanphill) [inline packages/editor/src/components/page-attributes/parent.js:98]: Nitpick: we could use vanilla JS `[].map()` here instead of Lodash.

**Comment by adamsilverstein** [comment]
@youknowriad didn't get a chance to test this; overall seemed fine. Thanks for the improvement.

Thinking we should add a similar toggle to author selector here so that if the site has a small number of pages, we use a traditional HTML selector vs the searchable selector.

**Pull request merged**